sources/test.py

```diff
@@ -73,15 +79,27 @@
                             "Please choose 'small' or 'medium'."))
         sys.exit(1)
 
+    if not args.model in ["feature_detection", "semantic_segmentation"]:
+        utils.logger.error("Unsupported model. Please consider ")
+        utils.logger.utils(("'feature_detection' or 'semantic_segmentation'"))
+        sys.exit(1)
+
     # Data path and repository management
     dataset_repo = os.path.join(args.datapath, args.dataset)
-    testing_name = "testing_" + str(image_size)
-    os.makedirs(os.path.join(dataset_repo, testing_name, "images"), exist_ok=True)
-    testing_filename = os.path.join(dataset_repo, testing_name + '.json')
+    input_repo = os.path.join(dataset_repo, "input")
+    preprocessed_repo = str(image_size) + "_" + aggregate_value
+    preprocessed_path = os.path.join(dataset_repo, "preprocessed", preprocessed_repo)
+    testing_filename = os.path.join(preprocessed_path, "testing.json")
```

Comment:
What is the purpose of the `testing.json` file?

Reply:
The same as `training.json` and `validation.json`: initialize the `Dataset` object that contains all the dataset image characteristics. Even if it is the testing phase, we still have a directory full of thousands of images. And we need to centralize the information so as to read them carefully.

The "labels" key within this file appears as unnecessary, however we need to store dummy labels because of TensorFlow feeding process (labels are expected during image batching, see `define_batch` method in `model.py`). This behavior could certainly be improved, however it seems more accurate to me to open a new dedicated issue.